nana/week2/tugas2.js

```diff
@@ -0,0 +1,46 @@
+/* 
+    2.	Buatlah program searching nama yang dapat dibatasi jumlah outputnya yang 
+    menerapkan  callback function dengan data sebagai berikut:
+*/
+let names = [
+    "Abigail", "Alexandra", "Alison",
+    "Ammanda", "Angela", "Bella",
+    "Carol", "Caroline", "Carolyn",
+    "Deirdre", "Diana", "Elizabeth",
+    "Ella", "Faith", "Olivia", "Penelope"
+]
+  function search(str,num, callback) {
+      try{
+          validasiInput(str)
+          validasiNum(num)
+          callback(str,num, tampilkan)
+          
+        }catch(err){
+            console.log(err)
+        }
+  }
+  function tampilkanHasil(str, num, callback) {
+      let result = names.filter((name) => name.toLowerCase().indexOf(str.toLowerCase()) !== -1);
+      result = result.slice(0, num);
```

Comment:
this code is not recomended.

opsi 1.
const result = names.filter((name) => name.toLowerCase().indexOf(str.toLowerCase()) !== -1).slice(0, num);

opsi 2.
const filterName = names.filter((name) => name.toLowerCase().indexOf(str.toLowerCase()) !== -1);
const result = filterName.silce(0, num);

opsi 3.
your beautifull code.